src/net.h

```diff
@@ -333,6 +328,19 @@ class CConnman
     // Whether the node should be passed out in ForEach* callbacks
     static bool NodeFullyConnected(const CNode* pnode);
 
+    template <typename Callable>
```

Comment:
Nice! I like how this cleans up the searching functions. Maybe add a comment (for doxygen) that this returns the first (arbitrary) node that matches a certain predicate, or `nullptr` otherwise.